src/miner.cpp

```diff
@@ -996,12 +1004,24 @@ void BitcoinMiner(std::shared_ptr<CReserveScript> coinbaseScript, bool fProofOfS
                 pblock->mixHash = mix_hash;
             } else if (pblock->IsSha256D() && pblock->nTime >= Params().PowUpdateTimestamp()) {
                 uint256 midStateHash = pblock->GetSha256dMidstate();
-                while (nTries < nInnerLoopCount &&
-                       !CheckProofOfWork(pblock->GetSha256D(midStateHash), pblock->nBits,
-                                         Params().GetConsensus(), CBlockHeader::SHA256D_BLOCK)) {
-                    boost::this_thread::interruption_point();
-                    ++nTries;
-                    ++pblock->nNonce64;
+                bool success = false;
+                while (nMidTries < nMidLoopCount) {
+                    while (nTries < nInnerLoopCount &&
+                           !(success = CheckProofOfWork(pblock->GetSha256D(midStateHash), pblock->nBits,
+                                   Params().GetConsensus(), CBlockHeader::SHA256D_BLOCK))) {
+                        boost::this_thread::interruption_point();
+                        ++nTries;
+                        ++pblock->nNonce64;
```

Comment:
Since this essentially runs it much longer, I do have some concerns over this change; but I'm not 100% in disagreement with it.  I need to give it another look on the next pass.

Reply:
To be clear, though, running the loop longer is the main way I've dealt with the lock contention from setting up the block template and nonce.

Reply:
The original response was much different, until I read the change description again.  I just want to double check everything that passes through when it fully exits and returns.